packages/wallet/src/redux/protocols/direct-funding/readme.md

```diff
@@ -0,0 +1,25 @@
+# Direct Funding Protocol
+
+The direct funding protocol is one of the funding protocols, which means that the protocol progresses the channel from after the exchange of the PreFund setup states to where the participants are able to exchange PostFund states. The keeps track of adjudicator funding in order to track when it is safe to deposit funds.
+
+## State Machine
+
+The protocol implements the following state machine:
+
+TODO: add action names for transitions.
+
+```mermaid
+graph LR
+ST((Start))-->NSD(NotSafeToDeposit)
+NSD-->|FUNDING_RECEIVED|TSSM(TransactionSubmissionStateMachine)
+ST-->TSSM
+TSSM-->WFFC(WaitForFundingConfirmation)
```

Comment:
This is probably already planned but wanted to leave a note that you'll need to handle the transaction-submission `Failure` case here too.

Reply:
Agreed. In the short term, how should direct funding handle transaction failure? I think some of the options are:

1. Abandon direct funding by returning a terminal funding failure state.
2. Retry the transaction.

In the long term, direct funding should probably offer both options to the user.

Reply:
I think Option 1 should be ok for the time being.